mirrord/protocol/src/tcp.rs

```diff
@@ -594,34 +597,44 @@ pub type ReceiverStreamBody = StreamBody<ReceiverStream<hyper::Result<Frame<Byte
 pub enum HttpResponseFallback {
     Framed(HttpResponse<InternalHttpBody>),
     Fallback(HttpResponse<Vec<u8>>),
-    Streamed(HttpResponse<ReceiverStreamBody>),
+
+    /// Holds the [`HttpResponse`] that we're supposed to send back to the agent.
+    ///
+    /// It also holds the original http request [`HttpRequestFallback`], so we can retry
+    /// if our hyper server sent us a
+    /// [`RST_STREAM`](https://docs.rs/h2/latest/h2/struct.Error.html#method.is_reset).
+    Streamed(
+        HttpResponse<ReceiverStreamBody>,
+        Option<HttpRequestFallback>,
```

Comment:
`Option` can be removed

Reply:
```
pub(crate) async fn handle_client_message(&mut self, message: LayerTcpSteal) -> Result<()> {
    (...)
    self.http_response(HttpResponseFallback::Streamed(http_response, None)).await?;
```

We don't have access to the `request` there, that's what forced me to use `Option`.

Reply:
Huh, what the hell. I removed the option locally and managed to make it comile :thinking:

Reply:
Ok, I was probably wrong and didn't notice :sweat_smile: 
Sorry